libutils/string_lib.c

```diff
@@ -593,6 +594,82 @@ long StringToLongExitOnError(const char *str)
     return result;
 }
 
+/**
+ * @brief Converts a string of numerals in base 10 to 64-bit signed int
+ *
+ * Result is stored in *value_out, return value should be checked.
+ * On error *value_out is unmodified and false is returned.
+ *
+ * @see StringToLong()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @param[out] value_out Where to store result on success, cannot be NULL
+ * @return true for success, false for error
+ */
+bool StringToInt64(const char *str, int64_t *value_out)
+{
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    int64_t val;
+
+    const int scanned = sscanf(str, "%" SCNd64 "", &val);
```

Comment:
Could use `strtoimax()` so that we get more details about the error (like from `strtol()` above).

Reply:
@vpodzime Since this function is for 64-bit integers, explicitly, it seems more correct to me to use the types and functionality the standard defines for 64-bit numbers. I guess you can rely on `int64_t == intmax_t` or implement additional range checking, but this seems wrong to me.

Reply:
Well, it's greater than or equal to `int64_t`, right? (could be static-asserted) And it would provide the same semantics and error checking and reporting as `strtol()`.

Reply:
pushed changes to make `StringToInt64()` match `StringToLong()` and blow up if `sizeof(intmax_t) != sizeof(int64_t)`